documentation/commands.md

```diff
@@ -1,233 +1,243 @@
+# Commands
+
 Commands in the form of JSON messages are used to start and stop file writing.
-Commands can be send through Kafka via the broker and topic specified by the
-`--command-uri` option.  Commands can also be given in the configuration
-file specified by `--config-file <file.json>` (see [commands in config file](#commands-can-be-given-in-the-configuration-file-as-well)).
 
-In the command, the `nexus_structure` defines the HDF hierarchy.
-The `nexus_structure` represents the HDF root object.  The following example
-shows how the HDF tree can be constructed using `children`.
-A child of type `stream` is a marker that a `HDFWriterModule` will insert
-the data into from a Kafka stream at that point in the hierarchy.  The options under
-the key `stream` specify the stream details containing, at least, the topic, the source name and
-the `HDFWriterModule` which should be used for writing.
-Depending on the `HDFWriterModule`, there will be more options specific to the
-`HDFWriterModule`.
+Commands are generally sent through Kafka via the broker and topic specified by the
+`--command-uri` option; however, commands can also be given in the [configuration
+file](Commands via the configuration file).
 
-#### Command to start writing a file
+## Command to start writing
 
-Further documentation:
+The start command consists of a number of parameters which are defined as key-value pairs in the JSON.
+These are:
 
-- [Groups](docs/groups.md)
-- [~~Datasets~~ documentation not yet written]()
-- [Attributes](docs/attributes.md)
-- [~~File Attributes~~ documentation not yet written]()
-- [~~Streams~~ documentation not yet written]()
+- cmd: The command name, must be `FileWriter_new`.
+- job_id: A unique identifier for the request, a UUID for example.
+- broker: The Kafka broker to use for data.
+- start_time: The start time in milliseconds (UNIX epoch) for data to be written from. Optional, if not supplied then the Kafka message time is used.
+- stop_time: The stop time in milliseconds (UNIX epoch) for data writing to stop. Optional, if not supplied then file writing continues until a stop command is received
+- service_id: The identifier for the instance of the file-writer that should handle this command. Optional, only needed if multiple file-writers present.
+- abort_on_uninitialised_stream: Whether to abort if the stream cannot be initialised. Optional, default is not to abort but carry on.
```

Comment:
What does "cannot be initialised" mean?

Reply:
:+1: It would be good to elaborate on that. Specifying an inaccessible broker in a stream URL, or a non-existent writer module are a couple of things which can cause this I think. There may be other possibilities.

Reply:
Basically anything that could go wrong: mistakes in the JSON, missing keys, no broker, multiple HDF issues...